feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/quicksettings/QuickSettingsScreen.kt

```diff
@@ -109,31 +117,47 @@ fun QuickSettingsScreenOverlay(
             }
         }
         BackHandler(onBack = onBack)
-        Column(
-            modifier =
-            modifier
-                .fillMaxSize()
-                .background(color = backgroundColor.value)
-                .alpha(alpha = contentAlpha.value)
-                .clickable(onClick = onBack),
-            verticalArrangement = Arrangement.Center,
-            horizontalAlignment = Alignment.CenterHorizontally
-        ) {
-            ExpandedQuickSettingsUi(
-                previewUiState = previewUiState,
-                currentCameraSettings = currentCameraSettings,
-                systemConstraints = systemConstraints,
-                shouldShowQuickSetting = shouldShowQuickSetting,
-                setVisibleQuickSetting = { enum: IsExpandedQuickSetting ->
-                    shouldShowQuickSetting = enum
-                },
-                onLensFaceClick = onLensFaceClick,
-                onFlashModeClick = onFlashModeClick,
-                onAspectRatioClick = onAspectRatioClick,
-                onCaptureModeClick = onCaptureModeClick,
-                onDynamicRangeClick = onDynamicRangeClick,
-                onLowLightBoostClick = onLowLightBoostClick
-            )
+
+        // Only needed to initialize
+        val currentDisplayRotation = LocalView.current.display.rotation
+        var newSurfaceRotation by remember { mutableIntStateOf(currentDisplayRotation) }
+        var prevSurfaceRotation by remember { mutableIntStateOf(currentDisplayRotation) }
+        key(LocalConfiguration.current) {
+            prevSurfaceRotation = newSurfaceRotation
+            newSurfaceRotation = LocalView.current.display.rotation
+        }
```

Comment:
I think this is assuming that the rotation of the display can only change when the `Configuration` also changes, but I don't think that's true in the general case. In free-form windowing, the device (and display) can rotate without the configuration changing.

Reply:
Can you recommend a way to test this? I'll try to address it.